modules/cache.js

```diff
@@ -96,32 +100,38 @@ Cache.prototype.quit = function (cb) {
  * This function will be triggered on new block, it will clear all cache entires.
  */
 
-Cache.prototype.onNewBlock = function () {
-	if(!self.isReady()) { return; }
-	async.map(['/api/blocks*', '/api/transactions*'], function (pattern) {
+Cache.prototype.onNewBlock = function (block, broadcast, cb) {
+	cb = cb ? cb: function () {};
```

Comment:
We only call callback in one place so `if (cb) { return cb(); }` is more readable. If we want to keep current approach `cb = cb || function () {};` is shorter and cleaner.